src/daft-scan/src/glob.rs

```diff
@@ -177,32 +176,30 @@ impl GlobScanOperator {
             }
             .into()),
         }?;
-        let mut partitioning_keys = if let Some(fp_col) = &file_path_column {
-            let partition_field =
-                PartitionField::new(Field::new(fp_col, DataType::Utf8), None, None)?;
-            vec![partition_field; 1]
-        } else {
-            vec![]
-        };
-        // If hive partitioning is set, extend the partition keys with  hive partition keys.
-        if hive_partitioning {
+        // If hive partitioning is set, create partition keys from the hive partition keys.
+        let (mut partitioning_keys, mut generated_fields) = if hive_partitioning {
             let hive_partitions = parse_hive_partitioning(&first_filepath)?;
             let hive_partition_schema = hive_partitions_to_schema(&hive_partitions)?;
             let hive_partition_schema = match user_provided_schema.clone() {
                 Some(hint) => hive_partition_schema.apply_hints(&hint)?,
                 None => hive_partition_schema,
             };
-            let hive_partitioning_keys = hive_partition_schema
-                .fields
+            let hive_partitioning_keys = (&hive_partition_schema.fields)
                 .into_iter()
-                .map(|(_, field)| PartitionField::new(field, None, None))
+                .map(|(_, field)| PartitionField::new(field.clone(), None, None))
                 .collect::<Result<Vec<_>, _>>()?;
-            partitioning_keys.extend(hive_partitioning_keys);
+            (hive_partitioning_keys, hive_partition_schema)
+        } else {
+            (vec![], Schema::empty())
+        };
+        // If file path column is set, extend the partition keys.
+        if let Some(fp_col) = &file_path_column {
+            let partition_field =
+                PartitionField::new(Field::new(fp_col, DataType::Utf8), None, None)?;
+            generated_fields = generated_fields
+                .non_distinct_union(&Schema::new(vec![partition_field.field.clone()])?);
```

Comment:
```suggestion
            generated_fields.fields.insert(partition_field.field.name.clone(), partition_field.field.clone());
```
Could you just append to the existing schema fields instead of needing to make a new schema and then doing non distinct union?

Reply:
Yep totally possible, shifted some stuff around to do this. Thank you again for all your attention!